jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -59,46 +77,120 @@
 	    logger.info("Using local cache for EJB timers: {}", USE_LOCAL_CACHE);
 	}
 
-	@SuppressWarnings("unchecked")
 	@Timeout
 	public void executeTimerJob(Timer timer) {
+        EjbTimerJob timerJob = (EjbTimerJob) timer.getInfo();
+        TimerJobInstance timerJobInstance = timerJob.getTimerJobInstance();
+        logger.debug("About to execute timer for job {}", timerJob);
 
-		EjbTimerJob timerJob = (EjbTimerJob) timer.getInfo();
-		logger.debug("About to execute timer for job {}", timerJob);
-		TimerJobInstance timerJobInstance = timerJob.getTimerJobInstance();
-		String timerServiceId = ((EjbGlobalJobHandle)timerJobInstance.getJobHandle()).getDeploymentId();
+        String timerServiceId = ((EjbGlobalJobHandle) timerJobInstance.getJobHandle()).getDeploymentId();
 
-		// handle overdue timers as ejb timer service might start before all deployments are ready
-		long time = 0;
+        // handle overdue timers as ejb timer service might start before all deployments are ready
+        long time = 0;
         while (TimerServiceRegistry.getInstance().get(timerServiceId) == null) {
-        	logger.debug("waiting for timer service to be available, elapsed time {} ms", time);
+            logger.debug("waiting for timer service to be available, elapsed time {} ms", time);
             try {
-				Thread.sleep(500);
-			} catch (InterruptedException e) {
-				e.printStackTrace();
-			}
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
             time += 500;
 
             if (time > OVERDUE_WAIT_TIME) {
-            	logger.debug("No timer service found after waiting {} ms", time);
-            	break;
+                logger.debug("No timer service found after waiting {} ms", time);
+                break;
             }
         }
-		try {
-			((Callable<Void>) timerJobInstance).call();
-		} catch (Exception e) {
-			logger.warn("Execution of time failed due to {}", e.getMessage(), e);
-			throw new RuntimeException(e);
+        try {
+            transaction(this::executeTimerJobInstance, timerJobInstance);
+        } catch (Exception e) {
+            recoverTimerJobInstance(timerJob, e);
 		}
 	}
 
+    private void executeTimerJobInstance(TimerJobInstance timerJobInstance) throws Exception {
+        try {
+            ((Callable<?>) timerJobInstance).call();
+        } catch (Exception e) {
+            logger.warn("Execution of time failed due to {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Exception e) {
+        // if we have next date fired means that it would have been reescheduled already by DefaultTimerJobInstance
+        if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
+            logger.warn("Execution of time failed Interval Trigger failed {}", ejbTimerJob.getTimerJobInstance());
+            return;
+        }
```

Comment:
This need some thought. I didn't allow that because overlapping. It gets a bit messy with for instance timer start events and I don't think add any value on these cases.
Also it would be very complicated to end user to understand what is going on.  It is not easy to follow through logging what is failing and what is not. 
Also the danger of increasing dramatically the number of triggers. if there is a network glitch like the one we are trying to solve here and we have interval triggers it will end up in multiple retrys
1) 1 interval trigger
2) retry #1 1 interval trigger
2) 2 interval trigger
3) retry #2 1 interval trigger 
4) 3 interval trigger
4) retry #3 1 interval trigger
4) retry #1 2 interval trigger

in the 5 step would be already a large amount of timers in the db.
and so on... this would grow exponentially.

Reply:
Well, will it really look like you wrote? I think that if the first trigger is not triggered because of an exception and let's say we implemented the retry also for reoccurring timers, then it won't fire `2 interval trigger` because the `1 interval trigger` did not succeed, which means it will first try to trigger the first instance and only after that succeeds it will move on to the second one. But I understand your point that we cannot save everything from application itself :) Maybe a little paragraph in documentation about how this behaves in such situation would be helpful for users.

Reply:
@MarianMacik  interval triggers by definition happens at fixed time so there is policy regarding delaying the next trigger because previous one has failed. At least I didn't see it that way.
your are presuming that way of work. let see that in case of interval trigger it won't retry the timer. I don't see any usefulness to that.